core/node/dns.go

```diff
@@ -0,0 +1,58 @@
+package node
+
+import (
+	"fmt"
+	"strings"
+
+	config "github.com/ipfs/go-ipfs-config"
+	doh "github.com/libp2p/go-doh-resolver"
+	madns "github.com/multiformats/go-multiaddr-dns"
+
+	"github.com/miekg/dns"
+)
+
+const ethResolverURL = "https://eth.link/dns-query"
+
+func newResolver(url string) (madns.BasicResolver, error) {
+	if !strings.HasPrefix(url, "https://") {
+		return nil, fmt.Errorf("invalid resolver url: %s", url)
+	}
+
+	return doh.NewResolver(url), nil
+}
+
+func DNSResolver(cfg *config.Config) (*madns.Resolver, error) {
+	var opts []madns.Option
+
+	hasEth := false
```

Comment:
Can we make this more generic and base  implicit defaults on a map that can grow/change over time?

```golang
map[string]string{
	"eth.": "https://eth.resolver.cloudflare-eth.com/dns-query",
	"crypto.": "https://eth.resolver.cloudflare-eth.com/dns-query",
}
```

I've added `.crypto` from [UnstoppableDomains](https://unstoppabledomains.com/)  because `eth.link` uses Cloudflare behind the scenes anyway :man_shrugging:  and Opera/Brave will support UD names out of the box too.

My thinking here is to use an endpoint that is as generic and "trusted" as possible, to minimize DNS leak caused by the initial clear text lookup for implicit DoH endpoint.

Reply:
Sure, that's easy to do.

Reply:
you probably meant to use `https://resolver.unstoppable.io/dns-query` for the `crypto.` resolver in the defaults -- I'll use that one.

Reply:
oh it's the same thing, nm.

Reply:
done.